datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -364,90 +393,60 @@ impl BatchCoalescer {
         Arc::clone(&self.schema)
     }
 
-    /// Add a batch, returning a batch if the target batch size or limit is reached
-    fn push_batch(&mut self, batch: RecordBatch) -> Result<Option<RecordBatch>> {
-        // discard empty batches
-        if batch.num_rows() == 0 {
-            return Ok(None);
-        }
+    fn push_batch(&mut self, batch: RecordBatch) -> CoalescerState {
```

Comment:
Could you please restore the documentation explaining what this does? Maybe saying to look at the documentation on `CoalescerState` to figure out what should be done on different return values